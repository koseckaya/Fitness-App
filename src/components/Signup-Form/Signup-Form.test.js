import React from 'react';
import renderer from 'react-test-renderer';
import  SignupForm from './signup-form';


const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUsedNavigate,
}));

    it('should render component with default props', () => {
        const tree = renderer
            .create(<SignupForm />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

