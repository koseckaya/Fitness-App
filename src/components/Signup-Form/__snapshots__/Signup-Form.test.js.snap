// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component with default props 1`] = `
<div
  className="form-wrap"
>
  <div
    className="form-container"
  >
    <h2
      className="form-title"
    >
      Track Your Progress & More!
    </h2>
    <form
      className="form__auth"
      onSubmit={[Function]}
    >
      <div
        className="mb-3"
      >
        <input
          aria-describedby="nameHelp"
          className="form-control__auth "
          id="name"
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="First Name"
          type="text"
        />
        <p
          className="invalid-feedback"
        />
      </div>
      <div
        className="mb-3"
      >
        <input
          aria-describedby="nameHelp"
          className="form-control__auth "
          id="lastName"
          name="lastName"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Last Name (optional)"
          type="text"
        />
        <p
          className="invalid-feedback"
        />
      </div>
      <div
        className="mb-3"
      >
        <input
          aria-describedby="emailHelp"
          className="form-control__auth "
          id="email"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Email address"
          type="email"
        />
        <p
          className="invalid-feedback"
        />
        <div
          className="form-text"
          id="emailHelp"
        >
          We'll never share your email with anyone else.
        </div>
      </div>
      <div
        className="mb-3"
      >
        <input
          className="form-control__auth "
          id="password"
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Password"
          type="password"
        />
        <p
          className="invalid-feedback"
        />
      </div>
      <div
        className="mb-3"
      >
        <input
          className="form-control__auth "
          id="confirmPassword"
          name="confirmPassword"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Re-enter password"
          type="password"
        />
        <p
          className="invalid-feedback"
        />
      </div>
      <div
        className="mb-3 form-check"
      >
        <input
          className="form-check-input "
          id="emailSubscription"
          name="emailSubscription"
          onBlur={[Function]}
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="form-check-label"
          htmlFor="emailSubscription"
        >
          Send me email notifications for new program launches, website or store updates (optional)
        </label>
      </div>
      <button
        className="button form-btn"
        disabled={true}
        type="submit"
      >
        Create account
      </button>
      <div
        className="form-delimiter"
      >
        or
      </div>
      <button
        className="button form-btn form-btn-google"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="form-google-icon"
          fill="#000000"
          height="30px"
          preserveAspectRatio="xMidYMid"
          viewBox="-3 0 262 262"
          width="30px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M255.878 133.451c0-10.734-.871-18.567-2.756-26.69H130.55v48.448h71.947c-1.45 12.04-9.283 30.172-26.69 42.356l-.244
 1.622 38.755 30.023 2.685.268c24.659-22.774 38.875-56.282 38.875-96.027"
              fill="#4285F4"
            />
            <path
              d="M130.55 261.1c35.248 0 64.839-11.605 86.453-31.622l-41.196-31.913c-11.024 7.688-25.82 13.055-45.257 13.055-34.523
 0-63.824-22.773-74.269-54.25l-1.531.13-40.298 31.187-.527 1.465C35.393 231.798 79.49 261.1 130.55 261.1"
              fill="#34A853"
            />
            <path
              d="M56.281 156.37c-2.756-8.123-4.351-16.827-4.351-25.82 0-8.994 1.595-17.697 4.206-25.82l-.073-1.73L15.26 
 71.312l-1.335.635C5.077 89.644 0 109.517 0 130.55s5.077 40.905 13.925 58.602l42.356-32.782"
              fill="#FBBC05"
            />
            <path
              d="M130.55 50.479c24.514 0 41.05 10.589 50.479 19.438l36.844-35.974C195.245 12.91 165.798 0 130.55 0 79.49 0 35.393 
 29.301 13.925 71.947l42.211 32.783c10.59-31.477 39.891-54.251 74.414-54.251"
              fill="#EB4335"
            />
          </g>
        </svg>
        Sign up with Google
      </button>
    </form>
  </div>
</div>
`;
